Refactor data

beechnut opened the pull request:
# Pull request summary

Overall, this PR:

- Refactors data into clearer organization, and fixes data references in all templates
- Removes unused layouts, partials, and data
- Adds `tags` collection and updates references to tags data
- Changes nearly all links from Jekyll format (`{{ site.baseurl }}`) to 11ty format (`{{ "path" | url }}`). Images and some navigation links remain.
- Updates site metadata  
- Cleans up some code, formatting, logic


Closes #3854 and #3859

cantsin commented:
We get some `Unexpected unnamed function` linter errors -- possible to disable this in the linter? (I think these are valid uses of anonymous functions)

cantsin reviewed (approved):
LGTM, it builds. would like to see the linting errors fixed but that's not strictly necessary.

beechnut commented on _data/guides.yml:
```diff
@@ -1,4 +1,6 @@
-#"promoted" is being used for layout, and per the design, it should only be set to true for two guides at a time
+# "promoted" is being used for layout, and per the design,
+# it should only be set to true for two guides at a time.
+# TODO: Create a validation to make sure it's only two
```
Listed in #3864 :)

beechnut commented on _includes/layouts/default.html:
```diff
@@ -1,15 +1,15 @@
----
-main: foo, bar, qaz
```
You're very welcome.

beechnut commented on config/collections.js:
```diff
@@ -0,0 +1,56 @@
+// However, that library requires ESM and this project is CommonJS.
+// The best thing to do would be to import the slugify filter, but I'm not quite
+// sure how to do that from this separate file — perhaps dependency injection
+// in .eleventy.js?
```
Done, added "look at all todos" to the epic

beechnut commented:
@cantsin I added "ignore" directives to the lines causing the eslint warnings. I also added a task to [post-migration tasks](https://github.com/18F/18f.gsa.gov/issues/3864) to remind us to double-check/un-ignore these eventually.
